aineko/core/dataset.py

```diff
@@ -35,140 +36,115 @@ class DatasetError(Exception):
-
-    Args:
-        dataset_name: Name of the dataset.
-        kafka_topic_to_future: Dictionary of kafka topics to futures.
-        status_list: List of status of dataset creation.
+        _future: Future representing the creation status of the dataset.
+
+    Usage:
+        ```python
+        dataset = MyDataset("my_dataset")
+        creation_status = dataset.create()
+        if creation_status.done():
+            print(f"Dataset {creation_status.dataset_name} has been created.")
+        else:
+            print(f"Dataset {creation_status.dataset_name} is being created.")
+        ```
     """
 
-    def __init__(
-        self,
-        dataset_name: str,
-        kafka_topic_to_future: Optional[Dict[str, Any]] = None,
-        status_list: Optional[List[Any]] = None,
-    ):
-        """Creation status of dataset or its components."""
+    def __init__(self, dataset_name: str, future: Optional[Future] = None):
+        """Initialize the dataset creation status."""
         self.dataset_name = dataset_name
-        self.kafka_topic_to_future = kafka_topic_to_future
-        self.status_list = status_list
+        self._future = future
 
     def done(self) -> bool:
-        """Return status of dataset creation.
-
-        For kafka topics, the status is represented by a dictionary
-        of kafka topics to futures. For kafka producers and consumers,
-        the status is represented by a list of status objects.
+        """Check if the dataset has been created.
 
         Returns:
-            True if all futures are done, otherwise False.
+            True if the dataset has been created, otherwise False.
         """
-        if not any([self.kafka_topic_to_future, self.status_list]):
-            return True
-        if self.kafka_topic_to_future:
-            return all(
-                future.done() for future in self.kafka_topic_to_future.values()
-            )
-        if self.status_list:
-            return all(status.done() for status in self.status_list)
-        return False
+        if self._future:
+            return self._future.done()
+        return True
 
 
 class AbstractDataset(abc.ABC, Generic[T]):
     """Base class for defining new synchronous Aineko datasets.
 
-    Subclass implementations can be instantiated using
-    the `from_config` method.
-
-    When defining a new dataset, the following methods must be implemented:
-
-    ```
-    - `read`
-    - `write`
-    - `create`
-    - `delete`
-    - `initialize`
-    - `exists`
-    ```
-
-    Example:
-    ```python
-    class MyDataset(AbstractDataset):
-        def read(self, **kwargs) -> Any:
-            pass
-
-        def write(self, **kwargs) -> Any:
-            pass
-
-        def create(self, **kwargs) -> Any:
-            pass
-
-        def delete(self, **kwargs) -> Any:
-            pass
+    Subclass implementations must implement all abstract methods. Please
+    refer to the documentation of each method for more information.
+    """
```

Comment:
I think it would be helpful to have a summary of datasets as a concept and an  overview of all the methods that needs to be implemented and why. For example:
```
A dataset comprises 2 subcomponents, the interface layer and the storage layer. The storage layer refers to the actual storage infrastructure that holds the data, and the interface layer is an API layer that allows for the interaction with the storage layer. 

__init__: This methods controls how the dataset is initialized when instantiated
create: Creation of the actual storage layer.
read: Read interaction to query data from the storage layer.
...
```